sled-hardware/src/illumos/partitions.rs

```diff
@@ -17,6 +21,82 @@ use illumos_utils::zpool::MockZpool as Zpool;
 #[cfg(not(test))]
 use illumos_utils::zpool::Zpool;
 
+/// NVMe devices use a meta size of 0 as we don't support writing addditional
+/// metadata
+static NVME_LBA_META_SIZE: u32 = 0;
+/// NVMe devices default to using 4k logical block addressing unless overriden.
+static DEFAULT_NVME_LBA_DATA_SIZE: u64 = 4096;
+
+/// NVMe device settings for a particular NVMe model.
+struct NvmeDeviceSettings {
+    /// The desired disk size for dealing with overprovisioning.
+    size: Option<u32>,
+    /// An override for the default 4k LBA formatting.
+    lba_data_size_override: Option<u64>,
```

Comment:
Nothing uses this today, so I am open to removing it from this PR in favor of not having dead code.